scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/benchmark/JavaBenchmark.java

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+
+package org.apache.mxnetexamples.javaapi.benchmark;
+
+import org.apache.mxnet.javaapi.Context;
+import org.kohsuke.args4j.CmdLineParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class JavaBenchmark {
+
+    private static boolean runBatch = false;
+
+    private static void parse(Object inst, String[] args) {
+        CmdLineParser parser  = new CmdLineParser(inst);
+        try {
+            parser.parseArgument(args);
+        } catch (Exception e) {
+            System.err.println(e.getMessage() + e);
+            parser.printUsage(System.err);
+            System.exit(1);
+        }
+    }
+
+    private static long percentile(int p, long[] seq) {
+        Arrays.sort(seq);
+        int k = (int) Math.ceil((seq.length - 1) * (p / 100.0));
+        return seq[k];
+    }
+
+    private static void printStatistics(long[] inferenceTimes, String metricsPrefix)  {
+
+        double p50 = percentile(50, inferenceTimes) / 1.0e6;
+        double p99 = percentile(99, inferenceTimes) / 1.0e6;
+        double p90 = percentile(90, inferenceTimes) / 1.0e6;
+        long sum = 0;
+        for (long time: inferenceTimes) sum += time;
+        double average = sum / (inferenceTimes.length * 1.0e6);
+
+        System.out.println(
+                String.format("\n%s_p99 %fms\n%s_p90 %fms\n%s_p50 %fms\n%s_average %1.2fms",
+                        metricsPrefix, p99, metricsPrefix, p90,
+                        metricsPrefix, p50, metricsPrefix, average)
+        );
+
+    }
+
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            System.out.println("Please specify model name");
+            return;
+        }
+        String modelName = args[1];
+        InferBase model;
+        if (modelName.equals("ObjectDetection")) {
+            runBatch = true;
+            ObjectDetectionBenchmark inst = new ObjectDetectionBenchmark();
+            parse(inst, args);
+            model = inst;
+        } else {
+            System.out.println("Model name not found! " + modelName);
+            return;
+        }
+        List<Context> context = new ArrayList<Context>();
+        if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
+                Integer.valueOf(System.getenv("SCALA_TEST_ON_GPU")) == 1) {
+            context.add(Context.gpu());
+        } else {
+            context.add(Context.cpu());
+        }
+
+        long[] result = new long[model.numRun];
+        model.preProcessModel(context);
+        if (runBatch) {
+            for (int i =0;i < model.numRun; i++) {
+                long currTime = System.nanoTime();
+                model.runBatchInference();
+                result[i] = System.nanoTime() - currTime;
+            }
+            printStatistics(result, modelName +"Batch");
+        }
+
+        ((ObjectDetectionBenchmark) model).batchSize = 1;
```

Comment:
I don't see this variable being used in the runSingleInference code path. Why are you setting it here?

Also, this could be a potential ClassCastException in the future.

Reply:
If you want to set class members from outside the class, use the Setter pattern. 
Do not access the members with the dot notation.